code/mps.h

```diff
@@ -120,6 +120,13 @@ typedef mps_addr_t (*mps_fmt_isfwd_t)(mps_addr_t);
 typedef void (*mps_fmt_pad_t)(mps_addr_t, size_t);
 typedef mps_addr_t (*mps_fmt_class_t)(mps_addr_t);
 
+/* Callbacks indicating that the arena has extended or contracted.
+ * These are used to register chunks with RtlInstallFunctionTableCallback
+ * <https://docs.microsoft.com/en-gb/windows/win32/api/winnt/nf-winnt-rtlinstallfunctiontablecallback>
+ * so that the client can unwind the stack through functions in the arena.
+ */
+typedef void (*mps_arena_extended_t)(mps_arena_t, mps_addr_t, size_t);
+typedef void (*mps_arena_contracted_t)(mps_arena_t, mps_addr_t, size_t);
```

Comment:
m. Second argument to both of these functions is documented as `void *` not `mps_addr_t`. The latter is mean to indicate managed memory, which this arguably is not. rule.generic.self

Reply:
Fixed: changed to void *